### PR TITLE
refactor(llm): drop deprecated prompting helpers, collapse persona renderer

### PR DESCRIPTION
**Pull request by 3aKHP**
## 动机

### CODE REVIEW

File: src/quickquip/llm/prompting.py [hunk 1]
```diff
@@ -50,57 +50,42 @@ def _compile_structured_persona(extras: dict[str, object]) -> str:
...
-            parts.append(f"当前情境：{identity['scenario']}")
-        if identity.get("self_reference"):
-            parts.append(f"自称方式：{identity['self_reference']}")
-        if parts:
-            sections.append("\n".join(parts))
+        sections.append(_render_persona_section(identity, {
+            "archetype": "角色原型",
+            "scenario": "当前情境",
+            "self_reference": "自称方式",
+        }))
```
[the remark this answers is not in the document]
Comment (3aKHP) [reply src/quickquip/llm/prompting.py:57]: Valid catch — this was a real regression, not just a docstring inaccuracy. Fixed in 0d85c3e.

You're right that my `isinstance(value, str) and value.strip()` guard silently dropped two classes of values the original `if X.get(...)` rendered: whitespace-only strings (`'   '` is truthy) and truthy non-string scalars (`1`, `True`). I over-tightened the guard during the refactor while claiming "zero behavior change" — that claim was wrong for this helper.

Restored to bare truthiness to truly match original behavior:
```python
value = table.get(key)
if value:
    parts.append(f"{label}：{value}")
```
The f-string stringifies non-strings exactly as the original per-field code did. Docstring rewritten to accurately state "truthiness guard" rather than the inaccurate "non-empty string" claim.

The new test `test_persona_identity_skips_missing_renders_truthy` now locks both edge cases you flagged: `archetype=1` (truthy int → rendered as "角色原型：1") and `scenario='   '` (truthy whitespace → rendered).

File: src/quickquip/llm/prompting.py [hunk 2]
```diff
@@ -50,57 +50,42 @@ def _compile_structured_persona(extras: dict[str, object]) -> str:
...
-            parts.append(f"核心渴望：{instinct['core_desire']}")
-        if instinct.get("stress_response"):
-            parts.append(f"压力反应：{instinct['stress_response']}")
-        if instinct.get("comfort_zone"):
-            parts.append(f"舒适区：{instinct['comfort_zone']}")
-        if parts:
-            sections.append("\n".join(parts))
+        sections.append(_render_persona_section(instinct, {
+            "core_desire": "核心渴望",
+            "stress_response": "压力反应",
```
[the remark this answers is not in the document]
Comment (3aKHP) [reply src/quickquip/llm/prompting.py:87]: Fixed in 0d85c3e — `syntax_rhythm` and `tone_shift` now use local variables (`syntax_rhythm = voice.get("syntax_rhythm"); if syntax_rhythm: ...`) instead of the get-then-direct-access double lookup.

I kept them inline rather than routing through `_render_persona_section` because the voice section also carries `verbal_habits`/`verbal_constraints` with their list/string branching — splitting just these two fields out to the helper while leaving the other two inline would fragment the section. The local-variable treatment matches what I did for `biography.origin` and `world.context`, giving consistent anti-pattern elimination across all simple fields.

File: src/quickquip/llm/prompting.py [hunk 3]
```diff
@@ -50,57 +50,42 @@ def _compile_structured_persona(extras: dict[str, object]) -> str:
...
+            "archetype": "角色原型",
+            "scenario": "当前情境",
+            "self_reference": "自称方式",
+        }))
 
     biography = extras.get("biography")
     if isinstance(biography, dict):
-        parts = []
+        parts: list[str] = []
         if biography.get("origin"):
```
[the remark this answers is not in the document]
Comment (3aKHP) [reply src/quickquip/llm/prompting.py]: Fixed in 0d85c3e. Both `biography.origin` and `world.context` now store the `.get()` result in a local variable and reuse it in the f-string, eliminating the double lookup.

File: tests/unit/llm/test_prompting.py [hunk 4]
```diff
@@ -362,53 +362,6 @@ def test_build_messages_images_attached():
     assert "img.png" in msgs[-1].image_urls
 
 
-# ---------------------------------------------------------------------------
-# Deprecated functions (backward-compat, remove in cleanup phase)
-# ---------------------------------------------------------------------------
-
-def test_build_user_message_content_quoted():
-    content = build_user_message_content(
-        prompt="这是什么？",
```
[the remark this answers is not in the document]
Comment (3aKHP) [reply tests/unit/llm/test_prompting.py:371]: Addressed in 0d85c3e. Renamed/rewrote the test as `test_persona_identity_skips_missing_renders_truthy` and added the two edge cases you flagged: `archetype=1` (truthy non-string) and `scenario='   '` (whitespace-only string). Both now assert the value is rendered, locking the truthiness-guard semantics now that MEDIUM is fixed to match original behavior.
